app/Gists/File.php

```diff
@@ -0,0 +1,64 @@
+<?php namespace Gistlog\Gists;
+
+use Illuminate\Contracts\Support\Arrayable;
+
+class File implements Arrayable
+{
+    public $name;
+    public $type;
+    public $language;
+    public $url;
+    public $size;
+    public $content;
+
+    /**
+     * @param array|ArrayAccess $rawFile
+     * @return File
+     */
+    public static function fromGitHub($rawFile)
+    {
+        $file = new self;
+
+        $file->name = $rawFile['filename'];
+        $file->type = $rawFile['type'];
+        $file->language = $rawFile['language'];
+        $file->url = $rawFile['raw_url'];
+        $file->size = $rawFile['size'];
+        $file->content = $rawFile['content'];
+
+        return $file;
+    }
+
+    /**
+     * @param array|ArrayAccess $rawFiles
+     * @return Collection
+     */
+    public static function multipleFromGitHub($rawFiles)
+    {
+        // Remove the blog post itself.
+        array_shift($rawFiles);
```

Comment:
Why not just default to the first Markdown file? It is probably going to be very uncommon for someone to include multiple markdown files in the same gist, and we can make a note about that somewhere saying "if your gist includes multiple markdown files, make sure the one containing the post comes first".

Seems like a good compromise between not forcing most people to think about the alphabetical order stuff and also avoiding any configuration or hard name requirements.